app/src/main/java/org/sopt/and/MainActivity.kt

```diff
@@ -30,18 +64,213 @@ class MainActivity : ComponentActivity() {
     }
 }
 
+
+
+fun EmailValidCheck(email: String): Boolean {
+    var isValid = false
+    //val expression = "^[\\w.-]+@([\\w\\-]+\\.)+[A-Z]{2,4}$"
+    val inputStr : CharSequence = email
+    val pattern = Patterns.EMAIL_ADDRESS
+    val matcher = pattern.matcher(inputStr)
+    if(matcher.matches()){
+        isValid = true
+    }
+    return isValid
+}
+
+fun PasswordValidCheck(password: String): Boolean {
+    val pattern = "^(?=.*[a-z])(?=.*[A-Z])(?=.*\\d)|(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#\$%^&*])|(?=.*[a-z])(?=.*\\d)(?=.*[!@#\$%^&*])|(?=.*[A-Z])(?=.*\\d)(?=.*[!@#\$%^&*]).{8,20}\$".toRegex()
+    return password.matches(pattern)
+
+}
+
 @Composable
-fun Greeting(name: String, modifier: Modifier = Modifier) {
-    Text(
-        text = "Hello $name!",
-        modifier = modifier
-    )
+fun Greeting(modifier: Modifier = Modifier) {
+
+    val context = LocalContext.current
+    
+    //이메일, 비밀번호 유효 여부에 따른 회원가입 가능여부 flag
+    var email_flag = 0;
+    var password_flag = 0; //8~20자 이내 조건 확인
+    var toastMessage = "";
+
+    var shouldShowPassword = remember {mutableStateOf(false)}
+    var shouldDisplayShow = remember {mutableStateOf(true)} //0이면 show 보이기, 1이면 hide 보이기
+
+
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+            .background(Color(0xFF1B1B1B))
+            .padding(25.dp),
+        //horizontalAlignment = Alignment.CenterHorizontally
+    ){
+        Spacer(modifier = Modifier.height(20.dp))
+        Text(
+            "회원가입",
+            modifier = Modifier.fillMaxWidth(),
+            textAlign = TextAlign.Center,
+            fontSize = 18.sp,
+            color = Color.White
+
+        )
+        Spacer(modifier = Modifier.weight(0.35f))
+
+        Text(
+            "이메일과 비밀번호만으로\nWavve를 즐길 수 있어요!",
+            color = Color.White,
+            fontSize = 21.sp
+        )
+
+        var EmailText = remember { mutableStateOf("") }
+        var PasswordText = remember { mutableStateOf("") }
```

Comment:
매번 `EmailText.value` 작성하시기 귀찮으실텐데 `by` 키워드를 사용하시면 바로 접근이 가능합니다 :)
```suggestion
        var EmailText by remember { mutableStateOf("") }
        var PasswordText by remember { mutableStateOf("") }
```